docker/.Dockerfile

```diff
@@ -0,0 +1,22 @@
+FROM node:18.12.1
+
+WORKDIR /cuttle/
+
+ENV NPM_CONFIG_LOGLEVEL=notice
+
+RUN apt-get update
+RUN apt-get clean
+
+# Install app dependencies
+# A wildcard is used to ensure both package.json AND package-lock.json are copied
+# where available (npm@5+)
+COPY package*.json .
```

Comment:
You copy this in the next line. Though, I will say, IME I like to do something like this

```Dockerfile
COPY package*.json
RUN npm ci
COPY . .
```
This allows you to rebuild the image without rebuilding depndencies if only the code changes

Reply:
Great point, thanks 👍🏼

Reply:
Yep - so the idea is that the layer cache is only invalidated if the filesystem is modified as a result of the COPY operation; since `node_modules` is `.docker-ignore`'d, it won't get overwritten in the container.

https://docs.docker.com/build/cache/

Of course, do check my work on this (an easy test is to just change the code (but not `package.json` or `package-lock.json` and re-run `docker build`, and then make sure that the build doesn't re-run the `npm ci` step). That said, I've done this in a number of repos to great success :)